### PR TITLE
Digitalocean task

### PR DESCRIPTION
**Pull request by stgleb**

### CODE REVIEW

File: pkg/jobs/digitalocean/digital_ocean.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	}
+
+	dropletRequest := &godo.DropletCreateRequest{
+		Name:              config.Name,
+		Region:            config.Region,
+		Size:              config.Size,
+		PrivateNetworking: true,
+		SSHKeys:           fingers,
+		Image: godo.DropletCreateImage{
+			Slug: "ubuntu-stable",
```
[the remark this answers is not in the document]
Comment (stgleb) [reply pkg/jobs/digitalocean/digital_ocean.go]: @egopher Ok, I'll pass this with config and align with node profile afterwards.

File: pkg/jobs/digitalocean/digital_ocean.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	Name         string
+	K8sVersion   string
+	Region       string
+	Size         string
+	Role         string // master/node
+	Image 		 string
+	Fingerprints []string
+
+	DropletTimeout time.Duration
+	CheckPeriod    time.Duration
```
Comment (saheienko) [inline pkg/jobs/digitalocean/digital_ocean.go]: Suppose, it should be job config, not a droplet.

File: pkg/jobs/digitalocean/digital_ocean.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+	tagService     TagService
+}
+
+type Config struct {
+	Name         string
+	K8sVersion   string
+	Region       string
+	Size         string
+	Role         string // master/node
+	Image 		 string
```
Comment (saheienko) [inline pkg/jobs/digitalocean/digital_ocean.go]: Please, run gofmt.

File: pkg/jobs/digitalocean/digital_ocean.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+	return &Job{
+		storage:        s,
+		dropletService: c.Droplets,
+		tagService:     c.Tags,
+	}
+}
+
+func (j *Job) CreateDroplet(config Config) error {
+	config.Name = config.Name + "-" + config.Role + "-" + strings.ToLower(util.RandomString(5))
```
Comment (saheienko) [inline pkg/jobs/digitalocean/digital_ocean.go]: It is better to use a function here.

File: pkg/jobs/digitalocean/digital_ocean.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+		},
+	}
+
+	tags := []string{"Kubernetes-Cluster", config.Name}
+
+	// Create
+	droplet, _, err := j.dropletService.Create(dropletRequest)
+
+	if err != nil {
+		return err
```
Comment (saheienko) [inline pkg/jobs/digitalocean/digital_ocean.go]: Please, wrap errors with some additional info.

File: pkg/jobs/digitalocean/digital_ocean.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+func (t *TokenSource) Token() (*oauth2.Token, error) {
+	token := &oauth2.Token{
+		AccessToken: t.AccessToken,
+	}
+	return token, nil
+}
+
+func getClient(credentials map[string]string) *godo.Client {
+	token := &TokenSource{
+		AccessToken: credentials["token"],
```
Comment (saheienko) [inline pkg/jobs/digitalocean/digital_ocean.go]: Please, check if this key exists, otherwise return an error.

Comment (saheienko) [reply pkg/jobs/digitalocean/digital_ocean.go]: Why do use a map for passing credentials if you have only a token here?

**Pull request merged**